pkg/infra/serverlock/serverlock.go

```diff
@@ -307,7 +307,13 @@ func (sl *ServerLockService) releaseLock(ctx context.Context, actionName string)
 	ctx, span := sl.tracer.Start(ctx, "ServerLockService.releaseLock")
 	defer span.End()
 
-	err := sl.SQLStore.WithDbSession(ctx, func(dbSession *db.Session) error {
+	// ensure clean up happens even if the context is cancelled
+	dbCtx := ctx
```

Comment:
curious, why don't you use `dbCtx := context.WithoutCancel(ctx)`?

Reply:
mainly because i didn't know about it. thanks!